lib/rules/no-plusplus.js

```diff
@@ -46,12 +46,28 @@ module.exports = {
                 if (allowInForAfterthought && node.parent.type === "ForStatement") {
                     return;
                 }
+
+                let fixerFunc;
+
+                if (node.parent.type !== "ReturnStatement") {
```

Comment:
I don't think this check is sufficient. It's possible to use the result of `++` and `--` outside of a return statement:

```js
bar = 0;
foo = bar++;
// now foo = 0

bar = 0;
foo = bar += 1;
// now foo = 1; the fix changed the behavior of the code
```

Instead, I would recommend only doing a fix when the parent is an `ExpressionStatement`:

```js
if (node.parent.type === "ExpressionStatement" || node.prefix) {
  // do a fix
}
```